### PR TITLE
Hud10837/layer view state

### PR DESCRIPTION
**Pull request by hud10837**
Awaiting completion of #223

Adds `getLayerViewState` to `GeoViewProxy` and `onLayerViewStateChanged` to the composable `MapView()`. Verified ad-hoc with this code:

```kotlin
    val arcGISMap by remember { mutableStateOf(ArcGISMap(BasemapStyle.ArcGISStreets)) }
    val mapViewProxy = remember { MapViewProxy() }
    MapView(
        modifier = Modifier.fillMaxSize(),
        arcGISMap = arcGISMap,
        mapViewProxy = mapViewProxy,
        onLayerViewStateChanged = {
            val layer = arcGISMap.basemap.value?.baseLayers?.firstOrNull()
            val layerViewState = layer?.let { it -> mapViewProxy.getLayerViewState(it) }
            Log.e("LayerViewState", "${it.layer.name} changed: ${it.layerViewState.status.first()}")
            Log.e("LayerViewState", "getLayerViewState: ${layer?.name} changed to ${layerViewState?.status?.firstOrNull()}")
        }
    )
```

### CODE REVIEW

File: toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/MapView.kt [hunk 1]
```diff
@@ -268,7 +272,8 @@ private fun MapViewEventHandler(
     onLongPress: ((LongPressEvent) -> Unit)?,
     onTwoPointerTap: ((TwoPointerTapEvent) -> Unit)?,
     onPan: ((PanChangeEvent) -> Unit)?,
-    onDrawStatusChanged: ((DrawStatus) -> Unit)?
+    onDrawStatusChanged: ((DrawStatus) -> Unit)?,
+    onLayerViewStateChanged: ((GeoView.GeoViewLayerViewStateChanged) -> Unit)?
```
Comment (gunt0001) [inline toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/MapView.kt:276]: Could you move this up above `onInteractingChanged`, so it matches the parameter order in the MapView() function.

**Review by gunt0001 — APPROVED** [review]
@hud10837 - looks good, one minor comment.

File: microapps/MapViewIdentifyApp/app/src/main/java/com/arcgismaps/toolkit/mapviewidentifyapp/screens/MainScreen.kt
Comment (hud10837) [inline microapps/MapViewIdentifyApp/app/src/main/java/com/arcgismaps/toolkit/mapviewidentifyapp/screens/MainScreen.kt:1]: This change went in recently, but `const` is not allowed with `Dp`, so this is a quick fix to add with this PR

**Review by puneet-pdx — APPROVED** [review]

**Comment by hud10837** [comment]
Thanks @gunt0001 @puneet-pdx

**Pull request merged**